--- a/qtbase/src/corelib/io/qlockfile_unix.cpp
+++ b/qtbase/src/corelib/io/qlockfile_unix.cpp
@@ -84,10 +84,10 @@ QT_BEGIN_NAMESPACE

 static QByteArray localHostName() // from QHostInfo::localHostName(), modified to return a QByteArray
 {
-    QByteArray hostName(512, Qt::Uninitialized);
-    if (gethostname(hostName.data(), hostName.size()) == -1)
+    char hostName[256];
+    if (gethostname(hostName, sizeof(hostName)) == -1)
         return QByteArray();
-    hostName.truncate(strlen(hostName.data()));
-    return hostName;
+    hostName[sizeof(hostName) - 1] = '\0';
+    return QByteArray(hostName);
 }

 // ### merge into qt_safe_write?
